sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -66,9 +66,21 @@ private[sql] class DefaultSource
       mode match {
         case SaveMode.Append =>
           sys.error(s"Append mode is not supported by ${this.getClass.getCanonicalName}")
-        case SaveMode.Overwrite =>
-          fs.delete(filesystemPath, true)
+        case SaveMode.Overwrite => {
+          try {
+            if (!fs.delete(filesystemPath, true)) {
+              throw new IOException(
+                s"Unable to clear output directory ${filesystemPath.toString} prior"
+                  + s" to INSERT OVERWRITE a JSON table:\n")
```

Comment:
Can we say "prior to writing to JSON file" since a user can reach this code path through `DataFrame.save`, which is not related to table operation.